plugin/evm/atomic/sync/atomic_sync_provider.go

```diff
@@ -0,0 +1,52 @@
+// (c) 2021-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package sync
+
+import (
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/snow/engine/snowman/block"
+
+	"github.com/ava-labs/coreth/core"
+	"github.com/ava-labs/coreth/plugin/evm/atomic/state"
+	"github.com/ava-labs/coreth/plugin/evm/sync"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+var _ sync.SummaryProvider = &AtomicSyncProvider{}
+
+type AtomicSyncProvider struct {
+	chain      *core.BlockChain
```

Comment:
you could define an interface `Chain` for this `chain` field with only `GetBlockByNumber` and `HasState`, so that it removes the dependency on the core package.

Reply:
unfortunately it goes deeper than that as `GetBlockByNumber` returns `types.Block` so we end-up importing the core package anyway. We can create an adapter and wrap the passed blockchain here, but I think having a separate pkg for sync should be good enough.

Reply:
The separate package `sync` should be good enough, but just to clarify `types` doesn't import `core` as far as I know.
In libevm, we will use the upstream `type` and this will be less of a concern.